src/realm/object-store/audit.mm

```diff
@@ -0,0 +1,1381 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#include <realm/object-store/audit.hpp>
+#include <realm/object-store/audit_serializer.hpp>
+
+#include <realm/object-store/impl/collection_change_builder.hpp>
+#include <realm/object-store/impl/collection_notifier.hpp>
+#include <realm/object-store/impl/realm_coordinator.hpp>
+#include <realm/object-store/object_schema.hpp>
+#include <realm/object-store/object_store.hpp>
+#include <realm/object-store/property.hpp>
+#include <realm/object-store/schema.hpp>
+#include <realm/object-store/sync/app.hpp>
+#include <realm/object-store/sync/sync_manager.hpp>
+#include <realm/object-store/sync/sync_session.hpp>
+#include <realm/object-store/sync/sync_user.hpp>
+#include <realm/object-store/util/scheduler.hpp>
+
+#include <realm/dictionary.hpp>
+#include <realm/sync/instruction_replication.hpp>
+#include <realm/sync/noinst/client_history_impl.hpp>
+#include <realm/table_view.hpp>
+#include <realm/util/file.hpp>
+#include <realm/util/logger.hpp>
+
+#include <dispatch/dispatch.h>
+#include <external/json/json.hpp>
+#include <external/mpark/variant.hpp>
+#include <sys/time.h>
+
+using namespace realm;
+
+namespace realm {
+static void to_json(nlohmann::json& j, Timestamp const& ts) noexcept
```

Comment:
These seemingly need to be static rather than in an anonymous namespace for ADL to work correctly.